internal/pkg/cli/storage_init.go

```diff
@@ -517,12 +522,24 @@ func (o *initStorageOpts) validateStorageLifecycle() error {
 // validateWorkloadNameWithLifecycle requires the workload to be in the workspace if the storage lifecycle is on workload-level.
 // Otherwise, it only caches whether the workload is present.
 func (o *initStorageOpts) validateWorkloadNameWithLifecycle() error {
+	if o.lifecycle == lifecycleEnvironmentLevel {
+		hasEnv, err := o.ws.HasEnvironment()
+		if err != nil {
+			return fmt.Errorf("check if environments are managed in the workspace: %w", err)
```

Comment:
nit: I wonder if "manage" could create confusion for folks
```suggestion
			return fmt.Errorf("check if environment manifests exist in the workspace: %w", err)
```